config/config.go

```diff
@@ -312,24 +317,17 @@ func loadService(v *viper.Viper) (configmodels.Service, error) {
 	return service, nil
 }
 
-// LoadReceiver loads a receiver config from v under the subkey receiverKey using the provided factories.
-func LoadReceiver(receiverKey string, v *viper.Viper, factories map[string]component.ReceiverFactoryBase) (configmodels.Receiver, error) {
+// LoadReceiver loads a receiver config from v under the subkey receiverKey using the factory corresponding
+// to the receiver type from receiverKey.
+func LoadReceiver(receiverKey string, v *viper.Viper, factory component.ReceiverFactoryBase) (configmodels.Receiver, error) {
```

Comment:
I don't understand how you know what receiver factory to pass to this function?  It looks like duplicate logic between determining the receiver factory and here to decode the type and name again.

Can we do better?

Reply:
I don't disagree but I don't know how to make it much better without significant reworking.

I think it would help if DecodeTypeAndName returned a struct that could be passed and had its own types instead of playing with strings that can easily get mixed up. That was discussed in a previous PR but was punted since it was related to the main issue.

The only thing I can think of is passing typeStr and fullName as pre-parsed parameters. Is that better?

Reply:
@bogdandrutu I agree with you, `LoadReceiver` signature looks confusing.

@jrcamp I think there is one more option. What if we rename this func to:
```go
func CreateAndLoadReceiver(
  receiverFullName string, 
  receiverConfig *viper.Viper, 
  factory component.ReceiverFactoryBase,
) (configmodels.Receiver, error)
```

You can then call `DecodeTypeAndName` in your code and call `CreateAndLoadReceiver` after that. I think`CreateAndLoadReceiver` signature looks more coherent.

Reply:
@tigrannajaryan thanks for the suggest. I took it except I made receiverConfig receiversConfig since it's the `receivers` section of the config file. Not sure that extra s will help anybody or not. :P

Reply:
Please check https://github.com/open-telemetry/opentelemetry-collector/pull/799 and decide together with @bogdandrutu 
I am happy with what you guys decide to do.

Reply:
Lets start with https://github.com/open-telemetry/opentelemetry-collector/pull/801 because that simplifies the requirements of the LoadReceiver signature.